addons/refuel/functions/fnc_connectNozzleAction.sqf

```diff
@@ -128,6 +128,8 @@ private _attachPosModel = _target worldToModel (ASLtoAGL _bestPosASL);
         };
 
         [_unit, _target, _nozzle, _endPosTestOffset] call FUNC(refuel);
+        _unit setVariable [QGVAR(tempFuel), nil];
+        [_unit, _nozzle] call FUNC(turnOn);
```

Comment:
Here should be `canTurnOn` check before `turnOn` call like it is in interaction menu.

Reply:
Should also communicate to the player why the muzzle didn't turn on too then.

Reply:
Then maybe it's easier to make sure that `FUNC(refuel)` manages all situations checked in `canTurnOn` and not call it :)